apps/language_server/lib/language_server/providers/folding_range/token_pairs.ex

```diff
@@ -0,0 +1,96 @@
+defmodule ElixirLS.LanguageServer.Providers.FoldingRange.TokenPairs do
+  @moduledoc """
+  """
+
+  alias ElixirLS.LanguageServer.Providers.FoldingRange
+
+  @token_pairs %{
+    "(": [:")"],
+    "[": [:"]"],
+    "{": [:"}"],
+    do: [:catch, :rescue, :after, :else, :end],
```

Comment:
for?

Reply:
Yep, good catch! Probably `case` too. I think there's a larger discussion of which tokens should be included as well. I just focused on the `Kernel.SpecialForms`.

Reply:
Fixed by: a8c54a2

Leaving this conversation unresolved for a potential discussion around what belongs in `@token_pairs`.

Reply:
Notable change here: 195acca

Turns out, I didn't have to enumerate all the different block identifiers. They were all special cases of the `{:block_identifier, _, _}` token. This made the logic much simpler.